frontend/src/scenes/settings/project/SessionRecordingSettings.tsx

```diff
@@ -217,6 +227,99 @@ export function ReplayAuthorizedDomains(): JSX.Element {
+    return [
+        {
+            label: 'any',
+            value: 'any',
+        },
+        ...multivariate.variants.map((variant) => {
+            return {
+                label: variant.key,
+                value: variant.key,
+            }
+        }),
+    ]
+}
+
+function LinkedFlagSelector(): JSX.Element | null {
+    const { updateCurrentTeam } = useActions(teamLogic)
+    const { currentTeam } = useValues(teamLogic)
+
+    const { hasAvailableFeature } = useValues(userLogic)
+    const { featureFlags } = useValues(featureFlagLogic)
+    const flagIsEnabled = featureFlags[FEATURE_FLAGS.SESSION_RECORDING_SAMPLING]
+    const featureFlagRecordingFeatureEnabled =
+        flagIsEnabled || hasAvailableFeature(AvailableFeature.REPLAY_FEATURE_FLAG_BASED_RECORDING)
+
+    const [selectedFlag, setSelectedFlag] = useState<FeatureFlagBasicType | null>(null)
+
+    if (!featureFlagRecordingFeatureEnabled) {
+        return null
+    }
+
+    return (
+        <>
+            <div className="flex flex-col space-y-2">
+                <LemonLabel className="text-base">Enable recordings using feature flag</LemonLabel>
+                <p>Linking a flag means that recordings will only be collected for users who have the flag enabled.</p>
+                <div className="flex flex-row justify-start">
+                    <FlagSelector
+                        value={currentTeam?.session_recording_linked_flag?.id ?? undefined}
+                        onChange={(id, key, flag) => {
+                            setSelectedFlag(flag)
+                            updateCurrentTeam({ session_recording_linked_flag: { id, key } })
+                        }}
+                    />
+                    {currentTeam?.session_recording_linked_flag && (
+                        <LemonButton
+                            className="ml-2"
+                            icon={<IconCancel />}
+                            size="small"
+                            type="secondary"
+                            onClick={() => updateCurrentTeam({ session_recording_linked_flag: null })}
+                            title="Clear selected flag"
+                        />
+                    )}
+                </div>
+                {isObject(selectedFlag?.filters.multivariate) && (
+                    <>
+                        <LemonLabel className="text-base">Link to a specific flag variant</LemonLabel>
+                        <p>
+                            This is a multi-variant flag. You can link to "any" variant of the flag, and recordings will
+                            start whenever the flag is enabled for a user.
+                        </p>
+                        <p>
+                            Alternatively, you can link to a specific variant of the flag, and recordings will only
+                            start when the user has that specific variant enabled.
+                        </p>
+                        <LemonSegmentedButton
```

Comment:
Based on the gif in the description I would make this a little longer if needs be. `fullWidth` might be overkill but `className="min-w-X"` could look better